build_tools/azure/upload_codecov.sh

```diff
@@ -13,4 +13,6 @@ coverage combine --append
 popd
 cp $TEST_DIR/.coverage $BUILD_REPOSITORY_LOCALPATH
 
-codecov --root $BUILD_REPOSITORY_LOCALPATH -t $CODECOV_TOKEN || echo "codecov upload failed"
+curl -Os https://uploader.codecov.io/latest/linux/codecov
+chmod +x codecov
+./codecov -t ${CODECOV_TOKEN} --rootDir $BUILD_REPOSITORY_LOCALPATH || echo "codecov upload failed"
```

Comment:
I think we should check that the codecov binary has not been corrupted since its last release using the GPG checker:

https://docs.codecov.com/docs/codecov-uploader#integrity-checking-the-uploader

Alternatively, we could choose a fixed version of the codecov binary (instead of curling the `/latest/` URL) and hardcode the expected SHA256 sum value in `build_tools/azure/upload_codecov.sh` and only check that with sha256sum. This would be slightly faster as we would not need to import the gpg key from keybase.io nor the checksum file and its signature each time. On the otherwise we would need to update the codecov uploader URL and the expected sha256 values from time to time.

Reply:
Yep, this is would be safer.